modules/api_gateway/main.tf

```diff
@@ -216,6 +206,7 @@ resource "aws_api_gateway_integration" "rest_api_route_integration" {
   http_method             = aws_api_gateway_method.rest_api_route_method[each.key].http_method
   integration_http_method = each.value["lambda_key"] != "" ? "POST" : each.value["method"]
   type                    = each.value["type"]
+  credentials             = aws_iam_role.invocation_role.arn
```

Comment:
This appears to cause a bug in the terraform provider https://github.com/hashicorp/terraform-provider-aws/issues/21907 which breaks the APIGW until apply is run a second time. Might be able to work around it by setting the role ahead of time.